clogMonitor/src/components/TimeRange.js

```diff
@@ -15,7 +16,73 @@ import { FormControl, FormHelperText, InputLabel, OutlinedInput, Stack } from '@
  * 
  * @returns {React.ElementType}
  */
+
+
 const TimeRange = ({ startTime, startChangeHandler, endTime, endChangeHandler, direction="column" }) => {
+
+    //DST ********************** begin
+    const hasDSTeffect = time => {
+        // check to see if DST is in effect in user's timezone 
+        const parsedDate = new Date(time)
+        const jan = new Date(parsedDate.getFullYear(), 0, 1); 
+        const jul = new Date(parsedDate.getFullYear(), 6, 1); 
+        // standard time offset
+        const stdOffset = Math.max(jan.getTimezoneOffset(), jul.getTimezoneOffset());
+        //check to see if DST is in effect in user's country
+        if(parsedDate.getTimezoneOffset() !== stdOffset) {
+            return true;
+        }
+        return false;
+    }
+    
+    const hasDSTconflict = time => {
+        // check to see if chosen date/time has DST conflict
+        const parsedDate = new Date(time)
+        let curHour = parsedDate.getHours();
+        let curMinute = parsedDate.getMinutes();
+        //check to see if chosen date/time is the start date/time of DST
+        const curMidnight = new Date(parsedDate.getFullYear(), parsedDate.getMonth(), parsedDate.getDate(), 0).getTimezoneOffset();
+        const cur3AM = new Date(parsedDate.getFullYear(), parsedDate.getMonth(), parsedDate.getDate(), 3).getTimezoneOffset();
+        if (curMidnight !== cur3AM && curHour === 1 && curMinute >= 0 && curMinute < 60){
+                return true;
+           }
+        return false;
+    }
+    
+    const [dst1, setDst1] = React.useState('');
+    const [openDst1, setOpenDst1] = React.useState(false);
+    
+    const handleDst1Change = (event) => {
+        setDst1(event.target.value);
+    };
+    const handleDst1Close = () => {
+        setOpenDst1(false);
+    };
+    const handleDst1Open = () => {
+        setOpenDst1(true);
+    };
+
+    const [dst2, setDst2] = React.useState('');
```

Comment:
dst1 and dst2 should be defined in LogEventsFilters so that there is access to them for sending the times to the API. Could you move them and also make it so that they are used during the sending of the times to the API?